File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        if not head:
            return 
        l = []
        curr = head
        while curr:
            l.append(curr.val)
            curr = curr.next
        if l[::-1]==l:
            return True
        else:
            return False

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
 import java.util.*;
class Solution {
    public boolean isPalindrome(ListNode head) {
        if(head.next==null) return true;
        Stack<Integer> stack = new Stack<>();
        ListNode curr = head;
        stack.push(curr.val);
        curr = curr.next;
        while(curr!=null){
            if(stack.peek() == curr.val){
                stack.pop();
            }
            else{
                stack.push(curr.val);
            }
            curr = curr.next;
        }
        return stack.isEmpty();
    }
}